### DISCUSSION TITLE
Packaged File Format - Status

### DISCUSSION THREAD

**Discussion by eldipa** [xournalpp/xournalpp#4802] [General] [answered]
Hi,  does anyone know what is the development status of the new file format (https://github.com/xournalpp/xournalpp/projects/2 ) ?

I'm being using Xournal++ since a few years ago and I love it.

Recently I'm working with a `.pdf` (~2MB) heavily annotated, with a `.xopp` of ~40MB. The size of the file is not a real issue but the time it takes to save it  it is.

With a SSD disk, the save takes ~10 secs. I hacked the source code to write a non-compressed `.xopp` and the time reduced to ~6 secs.

I understand that it acknowledged by the developers that the file format can be improved and there are a few tickets open with questions and ideas.

Unfortunately I couldn't find the branch or issue that it is tracking the development of the new format. I would like to help.

**Comment by eldipa** [comment]
For reference: https://github.com/xournalpp/xournalpp/issues/937

**Reply by Technius** [reply, marked as answer]
Progress is basically tracked in that issue. There hasn't been much progress beyond https://github.com/xournalpp/xournalpp/pull/3782 . I've been really busy recently (and probably will continue be in the forseeable future), so help would be greatly appreciated.

**Reply by eldipa** [reply]
Awesome, thanks @Technius I took a look at that ticket (and a few others) and I have a quite reasonable idea. I will write my proposal in #937  as it is where I think all the main discussion took / takes place.

And there is no need for excuses, working on an open source project is always  a second priority and sometimes one cannot invest time on it. I will do my best to present very "cooked" proposal and PRs.
